packages/cli/src/utils/api.js

```diff
@@ -364,6 +366,61 @@ const validateApp = async (definition) => {
   });
 };
 
+// TODO: docstring, fix ensureDir failure
+const downloadSourceZip = async () => {
+  const linkedAppConfig = await getLinkedAppConfig(undefined, false);
+  if (!linkedAppConfig.id) {
+    throw new Error(
+      `This project hasn't yet been associated with an existing Zapier integration.\n\nIf it's a brand new integration, run \`${colors.cyan(
+        'zapier register'
+      )}\`.\n\nIf this project already exists in your Zapier account, run \`${colors.cyan(
+        'zapier link'
+      )}\` instead.`
+    );
+  }
+
+  const credentials = await readCredentials(true);
+
+  const url = constants.ENDPOINT + `/apps/${linkedAppConfig.id}/latest/pull`
+
+  const requestOptions = {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json; charset=utf-8',
+      'User-Agent': `${constants.PACKAGE_NAME}/${constants.PACKAGE_VERSION}`,
+      'X-Requested-With': 'XMLHttpRequest',
+      'X-Deploy-Key' : credentials[constants.AUTH_KEY]
+    },
+  };
+  const response = await fetch(url, requestOptions);
+
+  if (!response.ok) {
+    throw new Error(`Got HTTP error, status: ${response.status}`);
+  }
+
+  const contentType = response.headers.get('content-type');
+  if (!contentType || !contentType.includes('application/zip')) {
+    throw new Error('The API did not return a ZIP file');
+  }
+
+  const sourceZipPath = constants.SOURCE_PATH;
+  const appDir = process.cwd();
+  const dirPath = path.resolve(appDir, constants.BUILD_DIR);
```

Comment:
This can overwrite an existing `build/source.zip`, which isn't ideal. It'd be better if we download it to a temporary directory (using `os.tmpdir()`).